### PR TITLE
Add two docs and diagrams on authentication in Kyma 2.0

### PR DESCRIPTION
**Pull request by kazydek**
**Description**

Changes proposed in this pull request:

- Added two docs and diagrams to describe the basic principles of authentication in Kyma 2.0.

**Related issue(s)**
See also https://github.com/kyma-project/kyma/issues/11521

### CODE REVIEW

File: docs/04-operation-guides/security/all-02-authentication-in-kyma.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+3. After successful authentication, the identity provider issues a JWT token that is stored in the browser session and used for all subsequent requests.
+
+### ID Tokens
+
+ ID Tokens are JSON Web Tokens (JWTs) signed by an OIDC-compliant identity provider and returned as part of the OAuth2 response that attest to the end user's identity.
+ An example decoded JWT looks as follows:
+
+```json
+{
+  "iss": "http://127.0.0.1:5556/dex",
```
Comment (kwiatekus) [inline docs/04-operation-guides/security/all-02-authentication-in-kyma.md:24]: sample issuear should not refer to dex

File: docs/04-operation-guides/security/all-01-access-kyma.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+## Kyma Dashboard
+
+The diagram shows the Kyma access flow using Kyma Dashboard.
+
+![Kyma Dashboard](assets/all-kyma-dashboard.svg)
+
+>**NOTE:** Kyma Dashboard is permission-aware so it only shows elements to which you have access as a logged-in user. The access is RBAC-based.
+
+1. Access Kyma Dashboard.
+2. If Kyma Dashboard does not find a JWT token in the browser session storage, it forwards the authentication request to your Open ID Connect (OIDC)-compliant identity provider.
```
Comment (kwiatekus) [inline docs/04-operation-guides/security/all-01-access-kyma.md:22]: I'm not sure if Busola uses web storage for tokens. Need to be checked with @valentinvieriu

File: docs/04-operation-guides/security/all-02-authentication-in-kyma.md (hunk 1 above)
Comment (kwiatekus) [inline docs/04-operation-guides/security/all-02-authentication-in-kyma.md:24]:
```suggestion
  "iss": "https://myapp.eu.auth0.com",
```

**Review by kwiatekus — CHANGES_REQUESTED** [review]

File: docs/04-operation-guides/security/all-02-authentication-in-kyma.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+![Authentication diagram](./assets/all-authentication-diagram.svg)
+
+1. Access Kyma Dashboard.
+2. Kyma Dashboard redirects you to an OIDC-compliant identity provider to handle the authentication.
+3. After successful authentication, the identity provider issues a JWT token that is stored in the browser session and used for all subsequent requests.
+
+### ID Tokens
+
+ ID Tokens are JSON Web Tokens (JWTs) signed by an OIDC-compliant identity provider and returned as part of the OAuth2 response that attest to the end user's identity.
+ An example decoded JWT looks as follows:
```
Comment (klaudiagrz) [inline docs/04-operation-guides/security/all-02-authentication-in-kyma.md:18]:
```suggestion
 An example of the decoded JWT looks as follows:
```

File: docs/04-operation-guides/security/all-02-authentication-in-kyma.md (hunk 1 above)
Comment (kazydek) [reply docs/04-operation-guides/security/all-02-authentication-in-kyma.md:22]: Done :)

Comment (kazydek) [reply docs/04-operation-guides/security/all-02-authentication-in-kyma.md:22]: Thx, I changed it to the example you provided.

File: docs/04-operation-guides/security/all-02-authentication-in-kyma.md (hunk 3 above)
Comment (kazydek) [reply docs/04-operation-guides/security/all-02-authentication-in-kyma.md]: Done!

**Review by kwiatekus — APPROVED** [review]

File: docs/04-operation-guides/security/all-01-access-kyma.md (hunk 2 above)
Comment (Wawrzyn321) [reply docs/04-operation-guides/security/all-01-access-kyma.md:22]: Yup, we use `localStorage` for tokens and other auth data.
Technically, it doesn't have to be JWT. Access token or a pair of {"Client-Certificate-Data" and "Client-Key-Data"} will also do.

Comment (valentinvieriu) [reply docs/04-operation-guides/security/all-01-access-kyma.md:22]: Correct, we do not store IDTokens ( that are in JWT format ) in the local storage. We all the time request them from the OIDC provider

**Review by klaudiagrz — APPROVED** [review]
We'll need to correct the links to API Gateway docs but for now, I'm approving the PR

**Pull request merged**